okpt/test/steps/opensearch.py

```diff
@@ -0,0 +1,208 @@
+                 index_spec: Dict[str, Any]):
+    """Creates an OpenSearch index, applying the index settings/mappings.
+
+    Args:
+        es: An OpenSearch client.
+        index_name: Name of the OpenSearch index.
+        index_spec: Dictionary containing the index settings/mappings.
+
+    Returns:
+        An OpenSearch index creation response body.
+    """
+    return es.indices.create(index=index_name, body=index_spec)
+
+
+@profile.label('disable_refresh')
+@profile.took
+@profile.step
+def disable_refresh(es: elasticsearch.Elasticsearch):
+    """Disables the refresh interval for an OpenSearch index.
+
+    Args:
+        es: An OpenSearch client.
+
+    Returns:
+        An OpenSearch index settings update response body.
+    """
+    return es.indices.put_settings(body={'index': {'refresh_interval': -1}})
+
+
+@profile.label('bulk_add')
+@profile.step
+def bulk(es: elasticsearch.Elasticsearch, index_name: str, body):
+    return es.bulk(index=index_name, body=body)
+
+
+def bulk_index(es: elasticsearch.Elasticsearch, index_name: str,
+               partitions: Iterable[np.ndarray]):
+    """Creates an OpenSearch index, applying the index settings/mappings.
+
+    Args:
+        es: An OpenSearch client.
+        index_name: Name of the OpenSearch index to be indexed against.
+        partitions: An array of vector partitions to be indexed.
+
+    Returns:
+        An array of bulk injection responses.
+    """
+    results = []
+    for partition in partitions:
+        result = bulk(es=es, index_name=index_name, body=partition)
+        results.append(result)
+    return results
+
+
+@profile.label('refresh_index')
+@profile.took
+@profile.step
+def refresh_index(es: elasticsearch.Elasticsearch, index_name: str):
+    """Refreshes an OpenSearch index, making it available for searching.
+
+    Args:
+        es: An OpenSearch client.
+        index_name: Name of the OpenSearch index to be refreshed.
+
+    Returns:
+        An OpenSearch index refresh response body.
+    """
+    return es.indices.refresh(index=index_name)
+
+
+def delete_index(es: elasticsearch.Elasticsearch, index_name: str):
+    """Deletes an OpenSearch index.
+
+    Args:
+        es: An OpenSearch client.
+        index_name: Name of the OpenSearch index to be deleted.
+
+    Returns:
+        An OpenSearch index deletion response body.
+    """
+    es.indices.delete(index=index_name)
+
+
+@profile.label('query_index')
+@profile.step
+def query_index(es: elasticsearch.Elasticsearch, index_name: str,
+                body: Dict[str, Any]):
+    """Queries a vector against an OpenSearch index.
+
+    Args:
+        es: An OpenSearch client.
+        index_name: Name of the OpenSearch index to be searched against.
+        body: A dictionary containing the body of the search request.
+
+    Returns:
+        An OpenSearch query response body.
+    """
+    return es.search(index=index_name, body=body)
```

Comment:
How hard would it be to add extra options to this query? Such as excluding source from response?

Reply:
Will address this in a future PR. For now, since this command is defined in a `Step` class, the user can specify their own options to their query when they override `_action` method.